routers.go

```diff
@@ -113,59 +117,103 @@ func registerAPIHandlers(mux *router.Router, a storageAPI, w *webAPI) {
 	/// Object operations
 
 	// HeadObject
-	bucket.Methods("HEAD").Path("/{object:.+}").HandlerFunc(a.HeadObjectHandler)
+	bucket.Methods("HEAD").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.HeadObjectHandler(rw, r)
+	})
 	// PutObjectPart
-	bucket.Methods("PUT").Path("/{object:.+}").HandlerFunc(a.PutObjectPartHandler).Queries("partNumber", "{partNumber:[0-9]+}", "uploadId", "{uploadId:.*}")
+	bucket.Methods("PUT").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.PutObjectPartHandler(rw, r)
+	}).Queries("partNumber", "{partNumber:[0-9]+}", "uploadId", "{uploadId:.*}")
 	// ListObjectPxarts
-	bucket.Methods("GET").Path("/{object:.+}").HandlerFunc(a.ListObjectPartsHandler).Queries("uploadId", "{uploadId:.*}")
+	bucket.Methods("GET").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.ListObjectPartsHandler(rw, r)
+	}).Queries("uploadId", "{uploadId:.*}")
 	// CompleteMultipartUpload
-	bucket.Methods("POST").Path("/{object:.+}").HandlerFunc(a.CompleteMultipartUploadHandler).Queries("uploadId", "{uploadId:.*}")
+	bucket.Methods("POST").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.CompleteMultipartUploadHandler(rw, r)
+	}).Queries("uploadId", "{uploadId:.*}")
 	// NewMultipartUpload
-	bucket.Methods("POST").Path("/{object:.+}").HandlerFunc(a.NewMultipartUploadHandler).Queries("uploads", "")
+	bucket.Methods("POST").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.NewMultipartUploadHandler(rw, r)
+	}).Queries("uploads", "")
 	// AbortMultipartUpload
-	bucket.Methods("DELETE").Path("/{object:.+}").HandlerFunc(a.AbortMultipartUploadHandler).Queries("uploadId", "{uploadId:.*}")
+	bucket.Methods("DELETE").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.AbortMultipartUploadHandler(rw, r)
+	}).Queries("uploadId", "{uploadId:.*}")
 	// GetObject
-	bucket.Methods("GET").Path("/{object:.+}").HandlerFunc(a.GetObjectHandler)
+	bucket.Methods("GET").Path("/{object:.+}").HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		w.storageapi.GetObjectHandler(rw, r)
```

Comment:
Also 'rw' should be 'w' simpler since it's just a 'writer' - rw signifies that its 'readwrite' .  Also